scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -358,7 +358,8 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
           token.is[KwDo] || token.is[KwCatch] || token.is[KwFinally] || token.is[KwYield] ||
           token.is[KwMatch]
 
-        if (existingIndent == expected && canEndIndentation(curr.token)) {
+        // we also insert outdent if correct keyword is on the same line
+        if ((existingIndent == expected || expected == -1) && canEndIndentation(curr.token)) {
```

Comment:
With this additional rule, it seems that calling `countIndentAndNewline` isn't necessary at all.
It would be enough just to check that we are inside the indented region and keyword can close it.
Smth like:
```scala

sepRegions match {
  case region :: tail if region.isIndented && canEndIndetation(curr.token) =>
     // procude outdent
     true
   case _ => false
}
```

Reply:
Ach, you are right. This actually works in more situations that it didn't previously work like:
```
    val a = if false then
      1
      2 
       else
      3
```